### PR TITLE
Add Pure Storage FlashArray module to manage local user accounts

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
New module to manage local users on a Pure Storage FlashArray.
Allows configuration of user role and creation and change of an associated API token.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
storage/purestorage/purefa_user

##### ADDITIONAL INFORMATION
Ability to change a password is NOT IDEMPOTENT!!
We have to supply both old and new passwords for the change and if a playbook has already
been run and changed the password, running the same playbook again with cause this play to fail.

### CODE REVIEW

**Comment by genegr** [comment]
shipit

**Comment by raekins** [comment]
shipit

RON EKINS // Oracle Solutions Architect, Office of the CTO | Pure Storage, Inc
m: +44 (0)7772 808 498 <tel:+44+(0)7772+808+498> | ron@purestorage.com <mailto:ron@purestorage.com> | Twitter @RonEkins <https://twitter.com/intent/follow?original_referer=https%3A%2F%2Fpublish.twitter.com%2F&ref_src=twsrc%5Etfw&region=follow_link&screen_name=RonEkins&tw_p=followbutton>

File: lib/ansible/modules/storage/purestorage/purefa_user.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+description:
+- Create, modify or delete local users on a Pure Stoage FlashArray.
+author:
+- Simon Dodsley (@sdodsley)
+options:
+  state:
+    description:
+    - Create, delete or update `local user account
+    default: present
+    choices: [ absent, present, update ]
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefa_user.py]: What's the difference between

Seeing state `present` and state `update` looks a little strange here.

```yaml
- name: Change role type for existing user
  purefa_user:
    name: ansible
    role: storage_admin
    state: present # <----
    fb_url: 10.10.10.2
    api_token: e31060a7-21fc-e277-6240-25983c6c4592
```


```yaml
- name: Change role type for existing user
  purefa_user:
    name: ansible
    role: storage_admin
    state: update # <----
    fb_url: 10.10.10.2
    api_token: e31060a7-21fc-e277-6240-25983c6c4592
```

Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_user.py]: `present` is for the creation of a user, where you specify a specific role type. `update` can be used to change this, however, need to split this out into a separate section as `update` can also be used to change a password or force an API token change. Your example won't do anything as you create a user with `role: storage_admin` and then try and change to the same value. There is a check for this eventuality. 
The examples show creating auser with `readonly` and then a change to a different role.

Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_user.py]: Just fixed line 26 with a weird character in the description

Comment (gundalow) [reply lib/ansible/modules/storage/purestorage/purefa_user.py]: Generally for Ansible modules `state: present` will update the object to match what's in the task.

Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_user.py]: @gundalow I have refactored this to only use `present` and `absent`.

**Comment by genegr** [comment]
shipit

**Comment by gundalow** [comment]
Thanks, will be released in Ansible 2.8

**Pull request merged**